jetty-core/jetty-http/src/main/java/org/eclipse/jetty/http/MultiPartFormData.java

```diff
@@ -450,17 +461,17 @@ private class PartsListener extends MultiPart.AbstractPartsListener
             public void onPartContent(Content.Chunk chunk)
             {
                 ByteBuffer buffer = chunk.getByteBuffer();
+                long maxFileSize = getMaxFileSize();
+                fileSize += buffer.remaining();
+                if (maxFileSize >= 0 && fileSize > maxFileSize)
+                {
+                    onFailure(new IllegalStateException("max file size exceeded: %d".formatted(maxFileSize)));
+                    return;
+                }
+
```

Comment:
Moving this code here feels like assuming too much.
If the part has no file name, and useFilesForPartsWithoutFileName=false, we should not use maxFileSize, right?

Reply:
I concur with @sbordet that moving this code here looks a be wrong... but looking at the original code I see some strange stuff as well. 

Why are we keeping a `fileSize` and `memoryFileSize` field? Surely there should just be a single `partSize` and it might be considered in memory initially and then as a file later as it grows past a threshhold.
 
So it would make sense to update a single `partSize` field here, before checking file name.

If `isUserFilesForPartsWithoutFileName` is false, then we still need to check the max size of the part.   If it exceeds `maxMemoryFileSize` is that a failure?

Maybe the logic should be:
```java
    partSize += buffer.remaining();

    if (fileName != null || isUseFilesForPartsWithoutFileName())
    {
        if (partSize > maxFileSize)
        {
            onFailure(...)
            return;
         }
         if (partSize > maxMemoryFileSize)
         {
             // write to file
             return;
         }
    }
    else
    {
         if (partSize > maxMemoryFileSize)
         {
            onFailure(...)
            return;
         }
    }
```

Reply:
I think moving the code here is correct, and necessary.

Without this if we have a part in memory it has no size limit and will keep increasing until we run out of memory.

The setting `jakarta.servlet.MultipartConfigElement#getMaxFileSize` really means the max size for the content of a part. It doesn't mean if `getFileSizeThreshold() == -1` we should ignore `getMaxFileSize()` until we run out of memory.

Reply:
> The setting jakarta.servlet.MultipartConfigElement#getMaxFileSize really means the max size for the content of a part.

Where do you get this interpretation? The javadocs explicitly say "file uploads".

The total multipart size is controlled by `maxLength`, so yes if you configure them unlimited there are no limits.
Not sure if there are limits mandated by the spec either.

I'm not opposed to have stricter limits, but then it's just a default value change and we do not need to introduce new classes.